### ISSUE TITLE
Email messages sent from notifications or triggers does not support RTL

### ISSUE THREAD

**Issue by DjMagicFingers** [zammad/zammad#1834]
### Infos:

* Used Zammad version: 2.3.x
* Installation method (source, package, ..):
* Operating system: CentOS 7
* Database + version: postgres
* Elasticsearch version: 5.6
* Browser + version: various

### Expected behavior:

* Customers, agents and admins should receive email messages that originated in notifications or triggers that use RTL language like Hebrew with compliance to the origin of the source.
For example, a trigger that sends email message to agents or customers that has Hebrew text in it should be RTL and aligned to the right side of the message, keeping the text readable even if it had Latin characters within the sentence.

### Actual behavior:

* All email messages are received LTR, also causing signs and numbers that appear in the middle of the sentence to be un-understandable.

![email rtl messed up 13-3-2018](https://user-images.githubusercontent.com/32017425/36152129-862629b2-10d2-11e8-91cf-581e746698c3.png)

### Steps to reproduce the behavior:

*

Yes I'm sure this is a bug and no feature request or a general question.

(My wish is for Zammad to have a way to format and design email messages send from it per group or per organization - maybe in the future)

**Comment by mrflix** [comment]
Maybe we can automatically detect when the email content is RTL and then correctly set the direction. For example with this ruby library: https://github.com/waiting-for-dev/string-direction

Once detected we can set the `dir="rtl"` attribute in `<html>` tag of the HTML email.

I wasn't able to find out how to indicate the direction in the plain email text. Maybe email clients detect it on their own. I checked both the [basic email standard](https://tools.ietf.org/html/rfc2822) and the [MIME extension](https://en.wikipedia.org/wiki/MIME). There exists an [invisible character](https://en.wikipedia.org/wiki/Right-to-left_mark) that indicates RTL text.

**Comment by DjMagicFingers** [comment]
reading the detection options, it looks like it would be best to use marks rather than chars or others since an email message can contain both languages and characters - English (like email addresses etc') and Hebrew or Arab at the body of the message.
Using marks as described at the link you have sent makes sens - it would be reasonable to assume that an email message sent from an RTL enabled software or RTL enabled email client would already contain some kind of a Unicode mark for RTL.
As i am not a programmer, i would have to ask if the u200f or u200e marks are standard marks to be expected in every email message body?

**Comment by mrflix** [comment]
Hey MagicFingers,
where do the Hebrew translations for the notification and trigger emails come from? Did you write them yourself?

**Comment by DjMagicFingers** [comment]
I did.

**Comment by aliva** [comment]
I have manually edited `application_wrapper.html.erb` and changed  this line

```html
<body style="###html_email_css_font###">###html###</body>
```

to this

```html
<body style="text-align:right; direction:rtl; ###html_email_css_font###">###html###</body>
```

> this file is at `/opt/zammad/app/views/mailer/application_wrapper.html.erb` in deb installation

And now all emails are RTL.

> Note that if you are using this hacky fix you should update `application_wrapper.html.erb` after every update

**Comment by DjMagicFingers** [comment]
But what if I use multi-language environment?

I have half of my clients English speaking, non RTL clients that should get emails in English - LTR and text-align to the left, and the other halft of them use RTL languages - that should get their emails with RTL and text-align to the right.

In general, I think it would be best to have a button to set this up per ticket, even though, judging from the way Zammad handles incoming emails, it is doing a wonderful job understanding the language of INCOMING emails and formatting tickets accordingly (if only it could do the same for OUTGOING email it would be great!).

Here is an example of how it looks so everyone can better understand the need :):
![image](https://user-images.githubusercontent.com/32017425/83325305-0dd21800-a274-11ea-81e9-c8a6dd66aa07.png)

**Comment by mantas** [comment]
Hi @DjMagicFingers ,

Just to make sure, is this issue with notifications/triggers only? Or does it affect manually written responses as well?

**Comment by DjMagicFingers** [comment]
It affects all - auto response, manual response and notifications.



נשלח מסמארטפון ה-Samsung Galaxy שלי.

**Comment by DjMagicFingers** [comment]
I have noticed that there is no "he." email template file here:
/opt/zammad/app/views/mailer

Is that where i should customize emails sent from Zammad?

How can I add the relevant Hebrew language templates?

![image](https://user-images.githubusercontent.com/32017425/83364464-25abb800-a3aa-11ea-9e07-48d428713692.png)

![image](https://user-images.githubusercontent.com/32017425/83364447-07de5300-a3aa-11ea-8c56-8e2bcd1b27d8.png)

**Comment by DjMagicFingers** [comment]
Noticing the ticket was opened long time ago, the relevancy of the issue is for Zammad 3.3 and also latest Elasticsearch...
All the rest of the infos are the same.

**Comment by MrGeneration** [comment]
Please don't push issues with technical or development questions. This is the wrong place.
The better place to do so is the community at https://community.zammad.org

**Comment by DjMagicFingers** [comment]
> Please don't push issues with technical or development questions. This is the wrong place.
> The better place to do so is the community at https://community.zammad.org

OK, I will sent it over there, but I feel this is an issue (the system acts different than it should), not a request, not technical and not dev requests...

The system does know how to understand RTL direction when receiving emails that are RTL (has Hebrew characters in it for example, and the sender sent it as RTL) and it knows exactly and very accurately how to show them on the Zammad web interface but it does not work the other way around - meaning it does not style the outgoing messages correctly when the agent writes Hebrew or even when the User chooses Hebrew as interface language that does changes the whole interface to be RTL'd - all but outgoing messages...

**Comment by MrGeneration** [comment]
Yeah I got that with the RTL part which is covered by this issue, however - this comment https://github.com/zammad/zammad/issues/1834#issuecomment-636542402 is different.

We're working very hard on  fixing all issues and making Zammad even better than it is now. However, this takes time.

I'm limitting the conversation to contributors only on this issue, because it's getting quite noisy.
This is nothing personal against you but to ensure that we can focus on bug fixing.

I think this is in your favor as well. :)

**Comment by mgruner** [comment]
https://github.com/zammad/zammad/commit/c2793174b0f0a9608728e1e4c91fc737b4b19d3c perhaps this could be a solution here as well: using `dir="auto"`.

**Comment by mgruner** [comment]
@DjMagicFingers good news: a fix is in review and hopefully coming soon, based on `<html dir="auto">`. This should solve your use case, hopefully. Sorry that it took way too long!

Regarding management of the Hebrew translations and mail templates, please feel free to contribute to https://translations.zammad.org/projects/zammad/zammad-development/he_IL/. All notification templates can be translated there as well. The Hebrew translation could use some love, it's pretty outdated.

**Comment by mgruner** [comment]
This will be available with the upcoming Zammad version 6.3.
